### PR TITLE
Enable a few whitespace enforcing IDE rules

### PR DESCRIPTION
**Pull request by pranavkm**
Contributes to https://github.com/dotnet/aspnetcore/issues/24055

### CODE REVIEW

**Comment by davidfowl** [comment]
Whitespace rules? Is this gonna break the build for extra whitespace?

**Comment by pranavkm** [comment]
> Whitespace rules? Is this gonna break the build for extra whitespace?

There are a couple of rules in the repo that already enforce some of these: 

* https://github.com/dotnet/aspnetcore/blob/main/.editorconfig#L242-L243
* https://github.com/dotnet/aspnetcore/blob/main/.editorconfig#L262-L264

This is adding some additional formatting clauses around whitespace that the [roslyn repo uses](https://github.com/dotnet/roslyn/blob/main/.editorconfig#L290-L300).

File: .editorconfig [hunk 1]
```diff
@@ -55,6 +55,11 @@ csharp_style_namespace_declarations = file_scoped
 # Brace settings
 csharp_prefer_braces = true # Prefer curly braces even for one line of code
 
+# Whitespace options
+csharp_style_allow_embedded_statements_on_same_line_experimental = false
+csharp_style_allow_blank_lines_between_consecutive_braces_experimental = false
+csharp_style_allow_blank_line_after_colon_in_constructor_initializer_experimental = false
```
Comment (dougbu) [inline .editorconfig:61]: I don't know if @dotnet/aspdoi has discussed these settings but this particular one is actively annoying and I'd much rather we didn't take it.

Comment (dougbu) [inline .editorconfig:61]: In general, we should wait until experimental features have been released for real before incorporating them in this repo. There's no reason to rush here.
I'd go further there and say we're in the weeds here; #24055 was about FxCop, not StyleCop, and certainly not teensy whitespace rules.

**Pull request closed without merging**